falcon/routing/static.py

```diff
@@ -180,6 +195,19 @@ def match(self, path: str) -> bool:
             return path.startswith(self._prefix)
         return path.startswith(self._prefix) or path == self._prefix[:-1]
 
+    def generate_etag(self, file_path: str) -> falcon.ETag:
+        """Generate an ETag for a file under the file_path or fallback_filename."""
+        try:
+            etag = _generate_etag(file_path)
+        except IOError:
+            try:
+                if self._fallback_filename is None:
+                    raise falcon.HTTPNotFound()
+                etag = _generate_etag(self._fallback_filename)
```

Comment:
Same as above, we are already calling `fstat` and handling potential errors elsewhere in this file, we can just reuse the same result.